OrbitGl/PickingManager.cpp

```diff
@@ -83,20 +80,17 @@ bool PickingManager::IsDragging() const {
   return picked && picked->Draggable();
 }
 
-Color PickingManager::GetPickableColor(std::weak_ptr<Pickable> pickable, BatcherId batcher_id) {
-  PickingId id = GetOrCreatePickableId(pickable, batcher_id);
+Color PickingManager::GetPickableColor(std::shared_ptr<Pickable> pickable, BatcherId batcher_id) {
+  PickingId id = GetOrCreatePickableId(std::move(pickable), batcher_id);
   return ColorFromPickingID(id);
 }
 
 bool PickingManager::IsThisElementPicked(const Pickable* pickable) const {
-  auto picked = GetPicked().lock();
+  auto picked = GetPicked();
   return picked && picked.get() == pickable;
 }
 
 Color PickingManager::ColorFromPickingID(PickingId id) const {
-  static_assert(sizeof(PickingId) == 4 * sizeof(uint8_t),
-                "PickingId should be same size as 4 * uint8_t");
-  std::array<uint8_t, 4> color_values;
-  std::memcpy(&color_values[0], &id, sizeof(PickingId));
+  auto color_values = absl::bit_cast<std::array<uint8_t, 4>>(id);
```

Comment:
Can you check if this causes any warnings during compilation on Linux? I vaguely remember that there was a reason I left the memcpy() in one place, but I might be wrong.

Reply:
I do not see any warnings...